crates/crypto/src/hashes/sha256.rs

```diff
@@ -10,22 +10,23 @@ pub fn sha256(data: &[u8]) -> [u8; 32] {
         use ring::digest;
         let hash = digest::digest(&digest::SHA256, data);
 
-        // we're pretty sure this will always be 32 bytes long
-        assert!(
-            hash.as_ref().len() == digest::SHA256.output_len,
-            "ring's Sha256 implementation has returned a digest of len {}, expected 32",
-            hash.as_ref().len()
-        );
-
-        hash.as_ref().try_into().unwrap()
+        // Convert the digest to a fixed-size array
+        let mut array = [0u8; 32];
+        array.copy_from_slice(hash.as_ref());
+        array
     }
     #[cfg(not(feature = "ring"))]
     {
         // Only if "ring" is not enabled, but "sha2" is, does it use "sha2" for the sha256 impl.
         use sha2::Digest;
         let mut hasher = sha2::Sha256::new();
         hasher.update(data);
-        hasher.finalize().into()
+        let result = hasher.finalize();
+
+        // Convert the digest to a fixed-size array
+        let mut array = [0u8; 32];
+        array.copy_from_slice(&result);
+        array
```

Comment:
This is unnecessary, `hasher.finalize()` returns a type that can be safely converted `into()` a `[u8; 32]`.

Reply:
I went ahead and fixed this as per your recommendation, now the only change is adding test cases. Thanks for the feedback!